tests/unit/CMakeLists.txt

```diff
@@ -35,6 +35,7 @@ add_custom_target(run-unit-tests
     USES_TERMINAL
 )
 
+find_package(Python3 REQUIRED COMPONENTS Interpreter)
```

Comment:
Are you sure this is needed? I dont see the variables/targets defined by `find_package` being used in this file.

Normally, such transitive dependencies should be defined using the `PUBLIC` keyword in the `target_link_library`, so they propagate downstream.